File: 241227/최고의 33위치/best-place-of-33.py
n = int(input())

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

ans = 0

for i in range(n):
    for j in range(n):
        if i+2 >=n or j+2 >=n:
            continue
        
        cnt = 0
        for x in range(i, i+3):
            for y in range(j, j+3):
                cnt += grid[x][y]
        
        ans = max(ans, cnt)


print(ans)

File: 250319/최고의 33위치/best-place-of-33.cpp
#include <iostream>
#include <algorithm>
using namespace std;

#define MAX_N 20
#define INLINE 3

int n;
int grid[MAX_N][MAX_N];

bool InRange(int x, int y){
    return 0<=x && x<n && 0<=y && y<n;
}

int Calc(int x, int y) {
    int cnt = 0;

    for(int i=0; i<INLINE; i++) {
        for(int j=0; j<INLINE; j++) {
            if (InRange(x+i, y+j)) {
                cnt += grid[x+i][y+j];
            }
        }
    }

    return cnt;

}


int main() {
    cin >> n;

    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            cin >> grid[i][j];
        }
    }

    int ans = 0;

    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            ans = max(ans, Calc(i, j));
        }
    }

    cout << ans;

    
    return 0;
}